File: file-handling/handling-csv-files/python/csv-library/csv_library_append.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4

import csv  

def append_record(new_record):
    """Append a record to a CSV file"""
    # Initialise the filename

    filename = "movies_extra.csv";
    
    # Open the file in append mode using the with command
    with open(filename, mode="a") as file_object:
        # Use the csv library to write a new row to the file
        write_to_csv = csv.writer(file_object)
        write_to_csv.writerow(new_record)


def main():
    movie_one = ["The Batman", "2022", "176", "Action,Crime,Drama", "Matt Reeves", "8.3", "350"]
    append_record(movie_one)
    
    movie_two = ["Joker", "2019", "122", "Crime,Drama,Thriller", "Todd Phillips", "8.4", "335.5"]
    append_record(movie_two)


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == "__main__":
    main()

File: file-handling/handling-csv-files/c-sharp/csv-library/csv_library_append.cs
/*
Raspberry Pi Foundation
Developed to be used alongside Isaac Computer Science, part of the National Centre for Computing Education
Usage licensed under CC BY-SA 4

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.Collections.Generic;
using System.Globalization;
using System.IO;
using CsvHelper;
using CsvHelper.Configuration;

namespace IsaacCodeSamples
{
    // The Movie class holds information about a movie
    class Movie
    {
        public string Title { get; set; }
        public string Year { get; set; }
        public string Duration { get; set; }
        public string Genre { get; set; }
        public string Director { get; set; }
        public string Rating { get; set; }
        public string Revenue { get; set; }
    }

    
    class Program
    {
        // The Main method is the entry point for all C# programs
        public static void Main() {
            // Create a list of Movie objects for appending to the CSV file
            var movies = new List<Movie>() {
                new Movie {Title="The Batman", Year="2022", Duration="176", Genre="Action,Crime,Drama", Director="Matt Reeves", Rating="8.3", Revenue="350"},
                new Movie {Title="Joker", Year="2019", Duration="122", Genre="Crime,Drama,Thriller", Director="Todd Phillips", Rating="8.4", Revenue="335.5"}
            };            
            AppendRecord(movies);
        }

        // Append a record to a CSV file
        public static void AppendRecord(List<Movie> newRecords) {
            // Initialise the filename
            string filename = "movies_extra.csv";

            // Configure CsvHelper so it does not write the header record again
            var csvConfig = new CsvConfiguration(CultureInfo.CurrentCulture) {
                HasHeaderRecord = false
            };

            // Append to the CSV file using File, StreamWriter and CsvWriter
            using (var stream = File.Open(filename, FileMode.Append))
            using (var writer = new StreamWriter(stream))
            using (var csvWriter = new CsvWriter(writer, csvConfig)) {
                csvWriter.WriteRecords(newRecords);
            }
        }

        
    }
}